pkg/daemon/ceph/client/subvolumegroup.go

```diff
@@ -55,3 +57,35 @@ func DeleteCephFSSubVolumeGroup(context *clusterd.Context, clusterInfo *ClusterI
 	logger.Infof("successfully deleted cephfs %q subvolume group %q", volName, groupName)
 	return nil
 }
+
+// PinCephFSSubVolumeGroup pin the cephfs subvolume group
+func PinCephFSSubVolumeGroup(context *clusterd.Context, clusterInfo *ClusterInfo, volName string, cephFilesystemSubVolumeGroup *cephv1.CephFilesystemSubVolumeGroup, cephFilesystemSubVolumeGroupName string) error {
+	logger.Infof("pinning cephfs subvolume group %q of filesystem %q", cephFilesystemSubVolumeGroupName, volName)
+
+	args := []string{"fs", "subvolumegroup", "pin", volName, cephFilesystemSubVolumeGroupName}
+	if cephFilesystemSubVolumeGroup.Spec.Pinning.Distributed != nil {
+		setting := strconv.Itoa(*cephFilesystemSubVolumeGroup.Spec.Pinning.Distributed)
+		args = append(args, "distributed", setting)
+	} else if cephFilesystemSubVolumeGroup.Spec.Pinning.Export != nil {
+		setting := strconv.Itoa(*cephFilesystemSubVolumeGroup.Spec.Pinning.Export)
+		args = append(args, "export", setting)
+	} else if cephFilesystemSubVolumeGroup.Spec.Pinning.Random != nil {
+		setting := strconv.FormatFloat(*cephFilesystemSubVolumeGroup.Spec.Pinning.Random, 'f', -1, 64)
+		args = append(args, "random", setting)
+	} else {
+		// set by default value
+		args = append(args, "distributed", "1")
+	}
```

Comment:
I know that there is input validation on the CRD, but Rook should still validate that the user didn't set multiples in code . We never know if the validation might fail for some reason, or be disabled by a user. Or, today, they could use k8s v1.24, and then could set improper values. I would suggest adding a `Validate() error` method to the `cephFilesystemSubVolumeGroup.Spec.Pinning` struct in the API code that verifies the value ranges.

In general, we should always check that input values are acceptable in Rook code, even when we have CRD validations.

Reply:
Plus, I even already wrote code when I proposed this before, so you can largely just copy-paste, and then write unit tests: https://go.dev/play/p/SxlYzUyVayO

Reply:
Added, please check.

Reply:
Thank you